### ISSUE TITLE
Terraform Plan Hangs

### ISSUE THREAD

**Issue by ssubraman224** [terraform-providers/terraform-provider-aws#5819]
Terraform plan hangs abruptly without writing anything to the state file. 
I am using Terraform version -> terraform_0.11.8_linux_amd64.zip and provider version ->terraform-provider-aws_1.35.0.

Just keeps throwing the following ->  waiting for: "provider.aws" and 
"root", waiting for: "provider.aws (close)"

Please let me know if its an issue with some security feature or configuration? any ports to be kept open?

**Comment by bflad** [comment]
Hi @ssubraman224 👋  Sorry you are running into trouble.

Does this occur on earlier version of the AWS provider?

It would be great if we could get a [Gist](https://gist.github.com) with [debug logging enabled](https://www.terraform.io/docs/internals/debugging.html) so we can further troubleshoot. If you are worried about any sensitive data, it can be encrypted with the [HashiCorp GPG Key](https://keybase.io/hashicorp) or redacted as necessary.

> Please let me know if its an issue with some security feature or configuration? 

We would need to see the logs (and potentially an example of your provider configuration) to determine this.

> any ports to be kept open?

Terraform only makes outbound connections.

**Comment by ssubraman224** [comment]
Could u please tell us what ports and endpoints  does the terraform contact to successfully execute the scripts?

**Comment by bflad** [comment]
You will need to provide the Terraform configuration for us to determine that. That said, AWS has a [very large amount of endpoints](https://github.com/aws/aws-sdk-go/blob/master/aws/endpoints/defaults.go), which are all on port 443 (HTTPS) and _generally_ the hostnames are constructed by `SERVICE.REGION.DOMAIN` (e.g. `DOMAIN` is `amazonaws.com` for AWS Commercial and AWS GovCloud (US))

If you are trying to setup outbound firewall rules, you will likely want to reference this: https://docs.aws.amazon.com/general/latest/gr/aws-ip-ranges.html

**Comment by ssubraman224** [comment]
Right, we have opened the ports for the ones mentioned by AWS, our query is that is there any other port that needs to be kept open?? From terraform perspective??? like the ones for plugin downloads and anything of that sort??

**Comment by bflad** [comment]
I believe the only connections Terraform itself uses are https://releases.hashicorp.com/ (for provider downloads) and if enabled, https://checkpoint.hashicorp.com/ ([for update checks](https://www.terraform.io/docs/commands/index.html#upgrade-and-security-bulletin-checks)). All other connections will be handled via the providers themselves or provisioner configurations.

If its the AWS provider that is "hanging" on a connection somewhere, the AWS Go SDK logs in the debug logging should be indicating the problem near logging containing `[aws-sdk-go] DEBUG: Retrying Request`. You can also try lowering [`max_retries` in your provider configuration](https://www.terraform.io/docs/providers/aws/index.html#max_retries) to potentially get the AWS Go SDK errors to return faster.

**Comment by plainsane** [comment]
my plans are hanging at
```An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create
  ~ update in-place
-/+ destroy and then create replacement
 <= read (data resources)

Terraform will perform the following actions:

```

**Comment by bflad** [comment]
Hi @plainsane 👋 Are you able to get an error to return if you set this in your provider configuration(s)?

```
provider "aws" {
  # ... other configuration ...
  max_retries = 1
}
```

Otherwise, we will be looking for a [Gist](https://gist.github.com) with [debug logging enabled](https://www.terraform.io/docs/internals/debugging.html) so we can further troubleshoot.

**Comment by plainsane** [comment]
im sorry, our issues seem to be some whacky terraform that showed up on a merge...ill post more when i figure it out, but our main branch is working again.

to note, we were getting out of memory errors.  i feel the graph is cycling, but i have no idea, i just pulled that out of thin air.

**Comment by ssubraman224** [comment]
Can anyone help me configure the "fluentd" for sending logs to an instance through task definition? ECS?
